src/shared_modules/utils/windowsHelper.h

```diff
@@ -136,6 +169,54 @@ namespace Utils
         return ret;
     }
 
+
+    /* Reference: https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_2.6.0.pdf */
+    static std::string parseRawSmbios(const BYTE* rawData, const DWORD rawDataSize)
+    {
+        std::string serialNumber;
+        DWORD offset{0};
+
+        while (offset < rawDataSize && serialNumber.empty())
+        {
+            SMBIOSStructureHeader header{};
+            memcpy(&header, rawData + offset, sizeof(SMBIOSStructureHeader));
+
+            if (BASEBOARD_INFORMATION_TYPE == header.Type)
+            {
+                SMBIOSBasboardInfoStructure info{};
+                memcpy(&info, rawData + offset, sizeof(SMBIOSBasboardInfoStructure));
+                offset += info.FormattedAreaLength;
```

Comment:
Can we be sure that `FormattedAreaLength` will never be bigger than `rawDataSize` ? Because it's a value read from the array itself and it could have any value.

Consider the same for `offset += header.FormattedAreaLength;`